api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirDiagnosticReportServiceImpl.java

```diff
@@ -44,12 +46,24 @@ public DiagnosticReport saveDiagnosticReport(DiagnosticReport diagnosticReport)
 	
 	@Override
 	public DiagnosticReport updateDiagnosticReport(String uuid, DiagnosticReport diagnosticReport) {
+		if (diagnosticReport.getId() == null) {
+			throw new InvalidRequestException("Diagnostic Report resource is missing id.");
+		}
+		
+		if (diagnosticReport.getId() != uuid) {
```

Comment:
This should be `!diagnosticReport.getId().equals(uuid)`. Strings are objects and for objects `==` asserts that they refer to the same object. This works when the strings are all generated inside the VM because of Java's string interning but is likely to fail in a real-world situation.